### PR TITLE
Reply #100 BE Changes

### PR DESCRIPTION
**Pull request by maettuu**
Here's the implementation of the flow we agreed upon :)

1. host creates new session
2. the invite link is generated now using session_id directly
3. when scanning qr_code session details can be requested using the session_id
4. all information can be displayed and guest enters username with which the request to join is made
5. the endpoint to join a session is POST "/sessions/{session_id}/guests"

### CODE REVIEW

**Comment by danielamormocea** [comment]
Why do we need to have all session details for users that are not logged in?

File: server/app/controller.py [hunk 1]
```diff
@@ -18,63 +22,77 @@
...
+# @app.patch("/sessions/{session_id}/songs", status_code=status.HTTP_200_OK)
+# async def add_song(user_id: Annotated[str, Depends(service.verify_token)], session_id: str, song_id: str) -> Session:
+#     await service.validate_user(user_id)
+#     await service.validate_session(session_id)
+#     return await service.add_song_to_session(user_id, session_id, song_id)
+
+
+# TODO: this will be adapted once we have the postgres database
+@app.get("/songs/{pattern}", status_code=status.HTTP_200_OK, response_model=list[Song])
+async def get_songs(user_id: Annotated[str, Depends(service.verify_token)], pattern: str) -> list[Song]:
```
Comment (danielamormocea) [inline server/app/controller.py:76]: I'm confused - I put them already in my PR - why did you copy them?

**Review by danielamormocea — CHANGES_REQUESTED** [review]
It seems that a lot of the functionalities here were added 3 days ago (in PR#100), tackled in my PR - so I'm confused why these are duplicates.  This feels very wrong.

File: server/app/models/song.py
Comment (jemaie) [inline server/app/models/song.py:1]: why are all fields except artist(_id)s optional? lets make these also optional or remove optional from the others where not necessary

Comment (maettuu) [reply server/app/models/song.py:1]: because dana showed us when initializing lists, the optional parameter can be removed, is works perfectly fine :)

**Pull request merged**